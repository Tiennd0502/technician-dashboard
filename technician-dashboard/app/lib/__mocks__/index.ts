export * from './table';
export * from './categories';
export * from './services';
