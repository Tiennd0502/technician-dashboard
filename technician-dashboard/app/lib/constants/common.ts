export const LOGO_URL = '/images/technician.svg';

export const AVATAR_DEFAULT = '/images/empty-avatar.svg';

export const CURRENCY_CHARACTERS = {
  EURO: 'â‚¬',
};

export const TRUNCATE_STYLE = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
};

export const NO_DATA = 'No data found';
