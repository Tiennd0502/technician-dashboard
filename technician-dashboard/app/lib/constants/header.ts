import { AVATAR_DEFAULT } from '.';

export const HEADER_INFO = {
  user: {
    name: 'Belle',
    avatar: AVATAR_DEFAULT,
  },
  title: 'Product & Services',
};
