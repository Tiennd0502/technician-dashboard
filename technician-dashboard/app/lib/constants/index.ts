export * from './common';
export * from './filter';
export * from './sidebar';
export * from './header';
export * from './rules';
export * from './messages';
export * from './routes';
export * from './status';
export * from './table';
export * from './categories';
export * from './services';
