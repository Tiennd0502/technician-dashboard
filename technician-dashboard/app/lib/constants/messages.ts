export const ERROR_MESSAGE = {
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required.`,
};
