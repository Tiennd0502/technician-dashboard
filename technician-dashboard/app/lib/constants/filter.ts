import { SORT_TYPE } from '@/lib/interfaces';

export const PRODUCT_LIMIT = 5;

export const DEFAULT_PRODUCT_FILTER = {
  name: '',
  sortBy: 'name',
  order: SORT_TYPE.Asc,
};
