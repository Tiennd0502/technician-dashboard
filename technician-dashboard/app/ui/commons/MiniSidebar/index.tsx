'use client';

import { useCallback } from 'react';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { Flex, List, ListItem, Text, Button, useDisclosure } from '@chakra-ui/react';

import { Menu } from '@/lib/interfaces';
import { HamburgerIcon } from '@chakra-ui/icons';
import { Logo } from '..';

interface SidebarPros {
  listItem: Menu[];
}

const MiniSidebar = ({ listItem }: SidebarPros) => {
  const pathname = usePathname();
  const { isOpen, onToggle } = useDisclosure();

  return (
    <Flex pos='relative' flexDirection='column' bg='primary.500'>
      <Flex m='5' justifyContent='space-between' alignItems='center'>
        <Logo />
        <Button aria-label='Icon Toggle Menu' variant='outline' onClick={onToggle}>
          <HamburgerIcon />
        </Button>
      </Flex>
      {isOpen && (
        <List
          as={Flex}
          flexDirection='column'
          pos='absolute'
          zIndex='2'
          top='100%'
          bg='primary.500'
          w='100%'
          role='list'
        >
          {listItem?.map(({ id, label, leftIcon, href }, index) => {
            const LeftIcon = leftIcon || <></>;

            return (
              <ListItem
                key={id}
                _hover={{
                  background: 'background.component.tertiary',
                }}
                {...(pathname?.startsWith(href) && {
                  background: 'background.component.tertiary',
                })}
                {...(index === listItem.length - 1 && {
                  mt: 'auto',
                  mb: 2,
                })}
                onClick={onToggle}
              >
                <Flex as={Link} href={href} p='5' gap='2.5' _hover={{ textDecoration: 'none' }}>
                  <LeftIcon />
                  <Text color='text.tertiary'>{label}</Text>
                </Flex>
              </ListItem>
            );
          })}
        </List>
      )}
    </Flex>
  );
};

export default MiniSidebar;
