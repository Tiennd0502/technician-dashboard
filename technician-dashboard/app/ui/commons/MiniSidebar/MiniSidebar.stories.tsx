import type { Meta, StoryObj } from '@storybook/react';

import { MENU_ITEM_LIST } from '@/lib/constants';

import MiniSidebar from '.';

const meta: Meta<typeof MiniSidebar> = {
  title: 'Components/MiniSidebar',
  component: MiniSidebar,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof MiniSidebar>;

export const Default: Story = {
  args: { listItem: MENU_ITEM_LIST },
};
