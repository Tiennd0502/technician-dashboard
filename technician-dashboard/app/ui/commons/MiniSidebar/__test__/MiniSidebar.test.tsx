import { render } from '@testing-library/react';
import * as mediaQueryHooks from '@chakra-ui/react';

import { MENU_ITEM_LIST } from '@/lib/constants';
import MiniSidebar from '..';

const onToggle = jest.fn();

jest.mock('next/navigation', () => ({
  usePathname: jest.fn().mockReturnValue('/product-and-services'),
}));

jest.mock('@chakra-ui/react', () => {
  const originalModule = jest.requireActual('@chakra-ui/react');

  return {
    __esModule: true,
    ...originalModule,
    useDisclosure: jest
      .fn()
      .mockImplementation(() => ({ isOpen: true, onToggle: () => jest.fn() })),
  };
});

describe('MiniSidebar test cases', () => {
  jest.spyOn(mediaQueryHooks, 'useDisclosure').mockImplementationOnce(() => ({
    isOpen: false,
    onToggle: jest.fn(),
    onOpen: jest.fn(),
    onClose: jest.fn(),
    isControlled: true,
    getButtonProps: (props?: any) => jest.fn(),
    getDisclosureProps: (props?: any) => jest.fn(),
  }));

  const props = {
    listItem: MENU_ITEM_LIST,
  };

  test('should render correctly', () => {
    const { container } = render(<MiniSidebar {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render enough elements', () => {
    const { getByRole } = render(<MiniSidebar {...props} />);
    const menuList = getByRole('list');

    expect(menuList.childNodes).toHaveLength(MENU_ITEM_LIST.length);
  });

  test('Should render empty elements', () => {
    const { getByRole } = render(<MiniSidebar listItem={[]} />);
    const menuList = getByRole('list');

    expect(menuList.childNodes).toHaveLength(0);
  });
});
