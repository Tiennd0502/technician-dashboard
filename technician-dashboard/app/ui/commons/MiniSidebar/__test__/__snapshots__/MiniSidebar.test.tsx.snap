// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniSidebar test cases should render correctly 1`] = `
<div>
  <div
    class="css-1hrbm4s"
  >
    <div
      class="css-150u52e"
    >
      <a
        class="css-1owjrhv"
        href="/"
      >
        <img
          alt="Technician"
          data-nimg="1"
          decoding="async"
          height="38"
          loading="lazy"
          src="/images/technician.svg"
          style="color: transparent;"
          width="37"
        />
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-p3xeyl"
          >
            Technician
          </p>
          <p
            class="chakra-text css-1o9nuri"
          >
            At Home
          </p>
        </div>
      </a>
      <button
        aria-label="Icon Toggle Menu"
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
